apm-agent-core/src/main/java/co/elastic/apm/agent/bci/ElasticApmAgent.java

```diff
@@ -369,11 +369,15 @@ private static void logClassLoaderHierarchy(@Nullable ClassLoader classLoader, L
     /**
      * Reverts instrumentation of classes and re-transforms them to their state without the agent.
      * <p>
-     * This is only to be used for unit tests
+     * NOTE: THIS IS ONLY TO BE USED FOR UNIT TESTS
```

Comment:
> Do you see a problem with using this method to detach the agent in the future?

No, just thought of emphasizing this note 🙂 . However, before using that we should probably disable the tracer and make sure everything stops using it, because this may happen in bulks, so some code may be instrumented while other is not (a strange undefined state). 

> What about using 'are assertions enabled' heuristic and throw an exception if this code is used in production ?

Based on the above, we may DO start using it.